float32 data
float64 time
